tests/shared/core/training_data/story_reader/test_yaml_story_reader.py

```diff
@@ -552,6 +552,46 @@ def test_or_statement_if_not_training_mode():
     assert or_statement[1].intent["name"] == "intent2"
 
 
+def test_or_statement_with_slot_was_set():
+    stories = """
+    stories:
+    - story: tell name bob or joe
+      steps:
+      - intent: greet
+      - action: utter_greet
+      - intent: tell_name
+      - or:
+        - slot_was_set:
+            - name: joe
+        - slot_was_set:
+            - name: bob
+      - action: utter_generic_hi
+    """
+
+    reader = YAMLStoryReader(is_used_for_training=False)
```

Comment:
Can we have a test where ` is_used_for_training=True` so we can see the actual `or` unfolding? i.e how the output would look for actual training.